client/sites/components/sites-dashboard.tsx

```diff
@@ -183,30 +183,34 @@ const SitesDashboard = ( {
 					],
 			  }
 			: {} ),
-		type: selectedSite ? 'list' : 'table',
-		layout: {
-			styles: {
-				site: {
-					width: getSiteNameColWidth( isDesktop, isWide ),
-				},
-				plan: {
-					width: '126px',
-				},
-				status: {
-					width: '142px',
-				},
-				'last-publish': {
-					width: '146px',
-				},
-				stats: {
-					width: '106px',
-				},
-				actions: {
-					width: '74px',
-				},
-			},
-		},
-	} as View;
+		...( selectedSite
+			? { type: 'list', layout: {} }
+			: {
+					type: 'table',
+					layout: {
+						styles: {
+							site: {
+								width: getSiteNameColWidth( isDesktop, isWide ),
+							},
+							plan: {
+								width: '126px',
+							},
+							status: {
+								width: '142px',
+							},
+							'last-publish': {
+								width: '146px',
+							},
+							stats: {
+								width: '106px',
+							},
+							actions: {
+								width: '74px',
+							},
+						},
+					},
+			  } ),
+	};
 	const [ dataViewsState, setDataViewsState ] = useState< View >( defaultDataViewsState );
```

Comment:
I considered changing this from `View` to `ViewList | ViewTable`, since I'm pretty sure that's the only types of layouts we need to support.

However, `<DataViews>`'s `onChangeView` prop, defined in Gutenberg, always provides a `View`. There's no way to tell Gutenberg that the component should only support list or table layouts.

So since `<DataViews>` _might_ return any type of `View`, we need to support any type of `View`.